crypto/x509/v3_purp.c

```diff
@@ -625,9 +587,10 @@ static int check_purpose_ns_ssl_server(const X509_PURPOSE *xp, const X509 *x,
                                        int ca)
 {
     int ret;
+
     ret = check_purpose_ssl_server(xp, x, ca);
     if (!ret || ca)
-        return ret;
+        return 0;
```

Comment:
Is it possible that both `ret` and `ca` could be non-zero? In such a case, the return value would be non-zero.  The same thing at L617 and L682, below.

Reply:
I think it's not possible, but I'm not positive, to I reverted to ```return ret``` and updated the commit.  good point, thanks.